layer_utils.py

```diff
@@ -30,6 +30,52 @@ def index_points(points, idx):
     return new_points
 
 
+class PConvLinearOptFunction(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, input_feat, neighbor_inds, inverse_neighbors, inverse_k, inverse_idx,
+                        weightnet, additional_features, linear_weights, linear_bias):
+        neighbor_inds.requires_grad = False
+
+        output, pconv_output = pcf_cuda.pconv_linear_forward(
+            input_feat, neighbor_inds, weightnet, additional_features, 
+            linear_weights, linear_bias)
+
+        ctx.save_for_backward(input_feat, inverse_neighbors, inverse_k, inverse_idx, 
+                            neighbor_inds, weightnet, additional_features, 
+                            linear_weights, pconv_output)
+        return output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        saved = ctx.saved_tensors
+        input_feat, inverse_neighbors, inverse_k, inverse_idx, neighbor_inds, \
+        weightnet, additional_features, linear_weights, pconv_output = saved
+
+        grad_output = grad_output.contiguous()
+
+        grads = pcf_cuda.pconv_linear_opt_backward(
+            grad_output, input_feat, inverse_neighbors, inverse_k, 
+            inverse_idx, neighbor_inds, weightnet, additional_features,
+            linear_weights, pconv_output)
+
+        return grads[0], None, None, None, None, grads[1], grads[2], grads[3], grads[4]
+
+class PConvLinearOpt(torch.nn.Module):
```

Comment:
Same comment as above: 
Explaining what the cuda kernel does briefly in a couple of sentences would be good here. That way, people who don't want to dig into the kernel but want to know how this particular implementation is more optimized than the naive PointConv can go through it.